### DISCUSSION TITLE
Firmware upgrade from Nov 2022

### DISCUSSION THREAD

**Discussion by Philofive** [tbnobody/OpenDTU#1980] [Q&A] [answered]
Hi there!

Title says it all. Thing is, i have 2 complete identical ESP32 and nRF24L01+ Chips, one i have used for quite some time without any problems but  i wanted to update now without losing my working ESP32. What do i do? Using the second ESP32 of course, using the same pins.

Flashed the software, used the Configuration File Backup, used the pin management (this was kind of new for me), but it seems, that i got it right by trying the standard configuration . (Simply uploaded the nodemcu_esp32.json)
I think i got it right,  because
System - Info - Radio Information tells me, that the software found the nRf24L01+ Chip, everything else is completely identical to the old ESP32, and i double checked manually,  I also checked if the software still finds the chip, if i mess with the wiring, answer is no. So it seems correct.

BUT:
Everything stay at "0" on the new ESP, whereas the old ESP works just fine. (i also unplugged the old ESP, because i use the same DTU number, but to no success)

Where can my thinking mistake be? Was it not correct to use the config management for backup and restoring?

Edit: of course i also used "NRF24" under Device Manager

Edit2: There is a console now: This is what i get and i guess, it is unusual
16:35:07.692 > Interrupt received
16:35:07.712 > Interrupt received
16:35:07.839 > RX Period End
16:35:07.839 > All missing
16:35:07.839 > Nothing received, resend whole request
16:35:07.839 > TX AlarmData Channel: 3 --> 15 81 80 17 66 80 16 05 96 80 11 00 66 3E 30 95 00 00 00 00 00 00 00 00 D9 6A BF 
16:35:07.846 > Interrupt received
16:35:07.854 > Interrupt received
16:35:07.860 > Interrupt received
16:35:07.866 > Interrupt received

**Comment by ghost** [comment]
Hi,
I think I read somewhere [here](https://github.com/tbnobody/OpenDTU) that versions before xx.xx.2023 are too old for a normal OTA update.
You have to do that once [opendtu-generic_esp32.factory.bin](https://github.com/tbnobody/OpenDTU/releases/download/v24.5.6/opendtu-generic_esp32.factory.bin) upload to ESP32.
The partitioning has changed.
[Upgrade Partition](https://www.opendtu.solar/firmware/howto/upgrade_partition/)
[Flasch Tool page](https://www.opendtu.solar/firmware/flash_esp/#overview-of-pre-compiled-binaries)

**Reply by Philofive** [reply]
Hi, thx for the answer. I was probably too unspecific. I used a new ESP32 and flashed that, exactly as you pointed out, with

`esptool.py --port /dev/ttyUSB0 write_flash -z 0x0 opendtu-generic_esp32.factory.bin` and i used exactly the pages, you pointed out.

i then "restored" pin management and it seems working. But it doesn't

P.S. i updated my first postwith a console output

**Comment by ghost** [comment]
I am sorry, my mistake.

Before flashing the ...factory...bin, delete the entire memory of the ESP32 once.
This is easy with the [Espressif tool](https://espressif.github.io/esptool-js/).

**Reply by Philofive** [reply]
i think i did that before, but just in case.... Did it again. But no success

**Reply by Philofive** [reply, marked as answer]
solved via

https://github.com/tbnobody/OpenDTU/discussions/1918

**Comment by MetaChuh** [comment]
@Philofive 

glad you've got it working.

for your convenience for the next time:

you can use our [`Community WebFlasher`](https://solar.metacontrol.eu/opendtu_webinstall/) which automatically takes care of board type and address settings for both factory flash and upgrade/downgrade flash via usb, to any version you like from the selecton.
no install needed, just open it in chrome or edge.
as well as if using a community opendtu fusion board, there will be no need to adjust or upload any pin_mapping.json.

<img width="929" alt="Bildschirmfoto 2024-05-10 um 18 03 27" src="https://github.com/tbnobody/OpenDTU/assets/45773305/e18db9c0-6d20-4086-8c1f-6d03f7d4cf28">

greetings,
metachuh
